File: HW/HW7/hw7.java
import java.util.Arrays;

public class hw7{
    public static int minMeetingRooms(int[][] intervals) {  
        Arrays.sort(intervals, (m1, m2) -> m1[0] - m2[0]);
        int servers = 1;

        for (int i = 1; i < intervals.length; i++){
            int[] current = intervals[i];
            int[] previous = intervals[i - 1];

            if(current[0] < previous[1]){
                servers++;
            }
            else{
                continue;
            }
        }
        return servers;
    }
    public static void main(String[] args) {
        int[][] intervals = {{0,30}, {5,10}, {15,25}};
        int[][] intervals2 = {{0,1}, {1,2}, {2,3}};
        int[][] intervals3 = {{1,5}, {0,9}, {1,6}};
        int[][] intervals4 = {{1, 3}, {2, 5}, {3, 7}, {4, 9}, {6, 8}, {3, 7}};

        System.out.println((minMeetingRooms(intervals)));
        System.out.println((minMeetingRooms(intervals2)));
        System.out.println((minMeetingRooms(intervals3)));
        System.out.println((minMeetingRooms(intervals4)));
        
    }
}

File: HW/HW7/hw7.py
from typing import List

class hw7:
    def minMeetingRooms(self, intervals: List[List[int]]) -> int:
        intervals.sort(key=lambda x: x[0])
        servers = 1

        for i in range(1, len(intervals)):
            current = intervals[i]
            previous = intervals[i - 1]

            if current[0] < previous[1]:
                servers += 1
            else: 
                continue
        return servers

hw7 = hw7()
intervals = [[0,30], [5,10], [15,25]]
intervals2 = [[0,1], [1,2], [2,3]]
intervals3 = [[1,5], [0,9], [1,65]]
intervals4 = [[1,3], [2,5], [3,7], [4,9], [6,8], [3,7]]

print(hw7.minMeetingRooms(intervals))
print(hw7.minMeetingRooms(intervals2))
print(hw7.minMeetingRooms(intervals3))
print(hw7.minMeetingRooms(intervals4))
